src/sentry/static/sentry/app/main.jsx

```diff
@@ -7,17 +7,24 @@ import routes from 'app/routes';
 import {loadPreferencesState} from 'app/actionCreators/preferences';
 import * as tracing from 'app/utils/tracing';
 
+function updateTracingData () {
```

Comment:
Why would you make the tracing calls so explicit? You never need to create a transaction without a span.

Reply:
I didn't want to duplicate same 3 calls in 2 places. And making them explicit like this, unifies the behavior with setting the current route. It's not doable in tracing code, as we need to listen for the changes here.